compiler/noirc_frontend/src/hir_def/traits.rs

```diff
@@ -35,7 +35,7 @@ pub struct TraitType {
 /// Represents a trait in the type system. Each instance of this struct
 /// will be shared across all Type::Trait variants that represent
 /// the same trait.
-#[derive(Debug, Eq)]
+#[derive(Clone, Debug, Eq)]
```

Comment:
Linking back to https://github.com/noir-lang/noir/pull/3736 as Jake explicitly removed this derive. It's necessary for this PR however.

Reply:
Could you add a note in the doc comment about how it's discouraged to clone `Trait`s due to its size?

Reply:
Opted for cleaning out Clone instead